### PR TITLE
Save & Remove from favorites 

### PR DESCRIPTION
**Pull request by tomasmoose**
## Overview
Dont use popup to save @ collection when clicking favorites icon, just save to favorites or remove directly. 

Bonus
+ generic linting stuff
+ Cleanup logic
+ use `response.ok` instead of checking for **200** status code for favorite requests

## Testing instructions
1. Make sure you can save datasets & widgets to favorites
2. Make sure you can remove them also

There should never appear any collection popup! 

## Pivotal task
https://www.pivotaltracker.com/story/show/156256396

### CODE REVIEW

File: css/_base.scss [hunk 1]
```diff
@@ -162,6 +162,15 @@ input, button, select, textarea {
   cursor: move;
 }
 
+.favourite-button {
```
Comment (mbarrenechea) [inline css/_base.scss]: I don't like this here in the `base.scss`.... Should we create a `c-favourite-button`? What do you think?

File: layout/explore/explore-datasets/explore-datasets-actions/index.js [hunk 2]
```diff
@@ -8,5 +8,5 @@ export default connect(
   state => ({
     ...state.explore.map
   }),
-  actions
+  ...actions
```
Comment (mbarrenechea) [inline layout/explore/explore-datasets/explore-datasets-actions/index.js]: Are you sure about this??

File: components/favorites/ToggleFavorite.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+const mapStateToProps = state => ({
+  user: state.user,
+  favourites: parseFavourites(state)
+});
+
+const mapDispatchToProps = dispatch => ({
+  toggleFavourite: fav => dispatch(toggleFavourite(fav))
+});
+
+export default connect(mapStateToProps, mapDispatchToProps)(ToggleFavorite);
```
Comment (mbarrenechea) [inline components/favorites/ToggleFavorite.js]: Split this component into a container and a component. Like we are doing with all of the components

File: layout/explore/explore-datasets/explore-datasets-actions/index.js (hunk 2 above)
Comment (mbarrenechea) [inline layout/explore/explore-datasets/explore-datasets-actions/index.js]: Are you sure about this?

File: css/_base.scss (hunk 1 above)
Comment (mbarrenechea) [inline css/_base.scss]: Let's create a `c-favourite-button` instead of puting this here (_base.scss)... What do you think?

Comment (tomasmoose) [reply css/_base.scss]: makes sense, will move it out to its own file !

File: layout/explore/explore-datasets/explore-datasets-actions/index.js (hunk 2 above)
Comment (tomasmoose) [reply layout/explore/explore-datasets/explore-datasets-actions/index.js]: not so sure, fixing :D

File: css/_base.scss (hunk 1 above)
Comment (tomasmoose) [reply css/_base.scss]: its a bit messy, we are shadow applying styles on all places where the favorite button is, so still needed to update lots of files, but well... its working a bit better now

**Pull request closed without merging**